include/boost/beast/http/fields.hpp

```diff
@@ -486,6 +505,35 @@ class basic_fields
     void
     insert(field, string_view, std::nullptr_t) = delete;
 
+    /** Insert a field.
+
+        If one or more fields with the same name already exist,
+        the new field will be inserted after the last field with
+        the matching name, in serialization order.
+        The value can be an empty string.
+
+        @param name The field name.
+
+        @param name_string The literal text corresponding to the
+        field name. If `name != field::unknown`, then this value
+        must be equal to `to_string(name)` using a case-insensitive
+        comparison, otherwise the behavior is undefined.
+
+        @param value The value of the field, as a @ref boost::beast::string_view
+
+        @param ec Set to indicate what error occurred, if any.
+        - If the size of @c name_string exceeds @ref max_name_size, the
+            error code will be @ref error::header_field_name_too_large.
+        - If the size of @c value exceeds @ref max_value_size, the
+            error code will be @ref error::header_field_value_too_large.
+    */
+    void
+    insert(field name, string_view name_string,
+           string_view const& value, error_code& ec);
+
+    void
+    insert(field, string_view, std::nullptr_t, error_code& ec) = delete;
```

Comment:
Why is there a `std::nullptr_t` param? And where's the doc for this?

Reply:
prevent passing passing the literal 0: https://github.com/boostorg/beast/issues/2085